### PR TITLE
chore: io/ioutil change to package io and os

### LINKED ISSUE #6261: io/ioutil was deprecated since go 1.16
### Is your feature request related to a problem?

The `io/ioutil` package was deprecated since go1.16. ref: https://go.dev/doc/go1.16#ioutil

### Describe the feature you'd like

Replace `io/ioutil` package with proper API in either `io` or `os` package

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by fatelei**
### What problem does this PR solve?

Issue Number: close #6261 

### What is changed and how it works?

replace `io/ioutil` with `io` and `os` package.

### Release note 

```release-note
replace `io/ioutil` with `io` and `os` package.
```

### CODE REVIEW

**Comment by lance6716** [comment]
Hi, kindly remind that don't remove the code block of Release note section

it should be

````
### Release note <!-- bugfixes or new feature need a release note -->

```release-note
replace `io/ioutil` with `io` and `os` package.
```
````

rather than 

````
### Release note <!-- bugfixes or new feature need a release note -->

replace `io/ioutil` with `io` and `os` package.
````

File: cdc/redo/writer/file.go [hunk 1]
```diff
@@ -485,7 +484,7 @@ func (w *Writer) shouldRemoved(checkPointTs uint64, f os.FileInfo) (bool, error)
 }
 
 func (w *Writer) getShouldRemovedFiles(checkPointTs uint64) ([]os.FileInfo, error) {
-	files, err := ioutil.ReadDir(w.cfg.Dir)
+	files, err := os.ReadDir(w.cfg.Dir)
```
Comment (lance6716) [inline cdc/redo/writer/file.go:487]: maybe the `files` should also be renamed to `entries`

**Review by lance6716 — APPROVED** [review]

File: cdc/redo/reader/file.go [hunk 2]
```diff
@@ -165,7 +165,7 @@ func downLoadToLocal(ctx context.Context, dir string, s3storage storage.External
 }
 
 func openSelectedFiles(ctx context.Context, dir, fixedType string, startTs uint64, workerNum int) ([]io.ReadCloser, error) {
-	files, err := ioutil.ReadDir(dir)
+	files, err := os.ReadDir(dir)
```
Comment (amyangfei) [inline cdc/redo/reader/file.go:168]: `ioutil.WriteFile` in L159 can be replaced too.

**Review by amyangfei — COMMENTED** [review]
please resolve conflicts, rest lgtm

**Comment by fatelei** [comment]
> please resolve conflicts, rest lgtm

done

**Review by amyangfei — APPROVED** [review]

**Pull request merged**